tests/models/abiotic/test_energy_balance.py

```diff
@@ -142,47 +138,54 @@ def test_calculate_leaf_and_air_temperature(
     )
     from virtual_ecosystem.models.abiotic_simple.constants import AbioticSimpleConsts
 
-    layer_structure = fixture_core_components.layer_structure
-    # TODO: Occupied canopies - the plants model should populate the filled_canopies
-    #       index in the data at some point.
-    # filled_canopies = np.repeat([False, True, False], [1, 3, 10])
+    lyr_str = fixture_core_components.layer_structure
 
     result = calculate_leaf_and_air_temperature(
         data=dummy_climate_data,
         time_index=1,
-        layer_structure=layer_structure,
+        layer_structure=lyr_str,
         abiotic_constants=AbioticConsts(),
         abiotic_simple_constants=AbioticSimpleConsts(),
         core_constants=CoreConsts(),
     )
 
-    exp_air_temp = layer_structure.from_template()
-    t_vals = [30.0, 29.999969, 29.995439, 28.796977, 20.08797]
-    exp_air_temp.T[..., [0, 1, 2, 3, 11]] = t_vals
+    exp_air_temp = lyr_str.from_template()
+    exp_air_temp[lyr_str.index_filled_atmosphere] = np.array(
+        [30.0, 29.999969, 29.995439, 28.796977, 20.08797]
+    )[:, None]
+
+    exp_leaf_temp = lyr_str.from_template()
+    exp_leaf_temp[lyr_str.index_filled_canopy] = np.array(
+        [30.078613, 29.091601, 26.951191]
+    )[:, None]
 
-    exp_leaf_temp = layer_structure.from_template()
-    tl_vals = [30.078613, 29.091601, 26.951191]
-    exp_leaf_temp.T[..., [1, 2, 3]] = tl_vals
+    exp_vp = lyr_str.from_template()
+    exp_vp[lyr_str.index_filled_atmosphere] = np.array(
+        [0.14, 0.140323, 0.18372, 1.296359, 0.023795]
+    )[:, None]
 
-    exp_vp = layer_structure.from_template()
-    vp_vals = [0.14, 0.140323, 0.18372, 1.296359, 0.023795]
-    exp_vp.T[..., [0, 1, 2, 3, 11]] = vp_vals
+    exp_vpd = lyr_str.from_template()
+    exp_vpd[lyr_str.index_filled_atmosphere] = np.array(
+        [0.098781, 0.099009, 0.129644, 0.94264, 0.021697]
+    )[:, None]
 
-    exp_vpd = layer_structure.from_template()
-    vpd_vals = [0.098781, 0.099009, 0.129644, 0.94264, 0.021697]
-    exp_vpd.T[..., [0, 1, 2, 3, 11]] = vpd_vals
+    exp_gv = lyr_str.from_template()
+    exp_gv[lyr_str.index_filled_canopy] = np.array([0.203513, 0.202959, 0.202009])[
+        :, None
+    ]
 
-    exp_gv = layer_structure.from_template()
-    gv_vals = [0.203513, 0.202959, 0.202009]
-    exp_gv.T[..., [1, 2, 3]] = gv_vals
+    # TODO - flux layer index does not include above but these tests do - what is best.
+    flux_index = np.logical_or(lyr_str.index_flux_layers, lyr_str.index_above)
```

Comment:
I think the flux above could probably be set to nan in the function, just need to check that this does not cause any errors. So will become irrelevant here, I think the solution is ok for now